app/services/sets/product_set.rb

```diff
@@ -114,6 +114,8 @@ def create_variant(product, variant_attributes)
 
       variant = product.variants.create(variant_attributes)
 
+      return false unless variant.save
```

Comment:
Actually this one make spec `spec/services/sets/product_set_spec.rb:175` fails... 
@mkllnk what do you think?